### PR TITLE
Apply caption to lightbox 2.0

### PR DESCRIPTION
**Pull request by muxin**
Fix https://github.com/ampproject/amphtml/issues/9050

### CODE REVIEW

File: extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js [hunk 1]
```diff
@@ -133,16 +138,33 @@ export class AmpLightboxViewer extends AMP.BaseElement {
             const deepClone = !element.classList.contains(
                 'i-amphtml-element');
             const nodeToClone = element.cloneNode(deepClone);
+            nodeToClone.removeAttribute('on');
+            const descText = this.manager_.getDescription(element);
+            if (descText) {
+              nodeToClone.descriptionText = descText;
```
Comment (aghassemi) [inline extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js]: please put a TODO to store this and the lightboxid in a data structure other than the node itself.

File: extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js [hunk 2]
```diff
@@ -79,6 +82,8 @@ export class AmpLightboxViewer extends AMP.BaseElement {
     /** @private {?Element} */
     this.descriptionBox_ = null;
 
+    this.clonedLightboxableElements = [];
```
Comment (muxin) [inline extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js]: done

File: extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js [hunk 3]
```diff
@@ -133,16 +138,33 @@ export class AmpLightboxViewer extends AMP.BaseElement {
             const deepClone = !element.classList.contains(
                 'i-amphtml-element');
             const nodeToClone = element.cloneNode(deepClone);
+            nodeToClone.removeAttribute('on');
```
Comment (muxin) [inline extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js]: done

File: extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js (hunk 1 above)
Comment (muxin) [inline extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js]: done

File: extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js [hunk 4]
```diff
@@ -133,16 +138,33 @@ export class AmpLightboxViewer extends AMP.BaseElement {
...
+              nodeToClone.descriptionText = descText;
+            }
+            this.clonedLightboxableElements.push(nodeToClone);
             this.carousel_.appendChild(nodeToClone);
           });
         });
       });
 
       this.container_.appendChild(this.carousel_);
+      this.win.document.addEventListener(
```
Comment (muxin) [inline extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js]: done

File: extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js [hunk 5]
```diff
@@ -132,17 +138,36 @@ export class AmpLightboxViewer extends AMP.BaseElement {
...
+            // TODO(yuxichen): store descriptionText and lightboxItemId in a
+            // list other than the node itself
+            this.clonedLightboxableElements_.push(clonedNode);
+            this.carousel_.appendChild(clonedNode);
           });
         });
       });
 
       this.container_.appendChild(this.carousel_);
+      this.element.addEventListener(
```
Comment (aghassemi) [inline extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js]: this.carousel_.addEvent...

**Pull request merged**